Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -953,6 +954,96 @@ function CreateRestApiPollerResourceProperties($armResource, $templateContentCon
             }
         }
     }
+
+    # paging placeholder
+    Paging($armResource)
+
+    # stepInfo placeholder
+    $hasStepInfo = [bool]($armResource.properties.PSobject.Properties.name -match "stepInfo")
+    $stepIds = @()
+    if ($hasStepInfo) {
+        $hasNextSteps = [bool]($armResource.properties.stepInfo.PSobject.Properties.name -match "nextSteps")
+        if ($hasNextSteps) {
+            foreach ($step in $armResource.properties.stepInfo.nextSteps) {
+                $stepIds += $step.stepId
+            }
+        } else {
+            Write-Host "Warning: 'stepInfo' object is missing 'nextSteps' array."
+        }
+
+        if ($stepIds.Count -gt 0) {
+            $stepIdsString = $stepIds -join ', '
+            Write-Host "List of identified 'stepId' in 'stepInfo' are: $stepIdsString"
+        }
+    }
+
+    # stepCollectorConfigs placeholder
+    $hasStepCollectorConfigs = [bool]($armResource.properties.PSobject.Properties.name -match "stepCollectorConfigs")
+    if ($hasStepCollectorConfigs) {
+        foreach ($stepId in $stepIds) {
+            # Check if the stepId exists in the stepCollectorConfigs
+            if ($armResource.properties.stepCollectorConfigs.PSObject.Properties.Match($stepId)) {
+                ProcessPropertyPlaceholders -armResource $armResource -templateContentConnections $templateContentConnections -isOnlyObjectCheck $false -propertyObject $armResource.properties.stepCollectorConfigs.$stepId.request -propertyName 'apiEndpoint' -isInnerObject $true -innerObjectName 'request' -kindType $kindType -isSecret $false -isRequired $false -fileType $fileType -minLength 4 -isCreateArray $false
+            } else {
+                Write-Host "Warning: Step ID $stepId not found in stepCollectorConfigs."
+            }
+        }
+    }
+
+    QueryParameters($armResource)
+}
+
+function Paging($armResource) {
+    $hasPaging = [bool]($armResource.properties.PSobject.Properties.name -match "paging")
+    if ($hasPaging) 
+    {
+        $pagingProperties = $armResource.properties.paging.PSobject.Properties.name
+
+        # Iterate over each property of the paging object
+        foreach ($propertyName in $pagingProperties) {
+            ProcessPropertyPlaceholders -armResource $armResource `
+                -templateContentConnections $templateContentConnections `
+                -isOnlyObjectCheck $false `
+                -propertyObject $armResource.properties.paging `
+                -propertyName $propertyName `
+                -isInnerObject $true `
+                -innerObjectName 'paging' `
+                -kindType $kindType `
+                -isSecret $false `
+                -isRequired $false `
+                -fileType $fileType `
+                -minLength 4 `
+                -isCreateArray $false
+        }
+    }
+}
+
+function QueryParameters($armResource) {
```

Comment:
Ideally this method should be handled in a generic way and should be able to create the parameters node for any json object not only for the queryparameters